libraries/chain/include/eosio/chain/block_header_state.hpp

```diff
@@ -120,6 +120,32 @@ struct pending_block_header_state : public detail::block_header_state_common {
                                                                const vector<digest_type>& )>& validator )&&;
 };
 
+/**
+ *  @struct block_header_state_core
+ *
+ *  A data structure holding hotstuff core information
+ */
+struct block_header_state_core {
+   uint32_t last_final_block_height = 0; // the block height of the last irreversible
+                                         // (final) block.
+   std::optional<uint32_t> final_on_strong_qc_block_height; // the block height of
+                                                            // the block that would
+                                                            // become irreversible
+                                                            // if the associated block
+                                                            // header was to achieve a
+                                                            // strong QC.
```

Comment:
I think you could use longer lines (similar to line 119 above) so the comment would not be spread out on so many lines.

Reply:
Thanks. The shorter lines were ugly. Changed.